### PR TITLE
Set `$post->filter` in `update_post_cache()` - Ticket #50567

### PR DESCRIPTION
**Pull request by sybrew**
Trac ticket: https://core.trac.wordpress.org/ticket/50567

---
**This Pull Request is for code review only. Please keep all other discussion in the Trac ticket. Do not merge this Pull Request. See [GitHub Pull Requests for Code Review](https://make.wordpress.org/core/handbook/contribute/git/github-pull-requests-for-code-review/) in the Core Handbook for more details.**

### CODE REVIEW

**Comment by sybrew** [comment]
Not being used to PRs (I mostly push to master like a real pro 👀), I didn't know my commit push instantly merged with the PR, forgoing courtesy awaiting a response on [my Core ticket reply](https://core.trac.wordpress.org/ticket/50567#comment:20).

**Comment by peterwilsoncc** [comment]
> Not being used to PRs (I mostly push to master like a real pro 👀), I didn't know my commit push instantly merged with the PR, forgoing courtesy awaiting a response on [my Core ticket reply](https://core.trac.wordpress.org/ticket/50567#comment:20).

Don't worry, always feel free to push modifications to your own PRs. You need to be happy with your contributions as much as anyone else. If a committer catches something, they won't be shy making suggestions. As we both know ;) 

The logic check to ensure `raw` sanitization makes sense to me.

**Comment by peterwilsoncc** [comment]
@sybrew Do you have any experience writing unit tests in PHPUnit?

Re-reading the ticket, I noticed it mentioned that it would be good to have some tests in place to make sure committing this code doesn't break anything. I think it's a wise suggestion.

**Comment by sybrew** [comment]
@peterwilsoncc I do not have experience with PHPUnit. However, since the posts/query system is fundamental to WordPress, I believe all necessary query tests exist. It is why I brought it up in comment:11.

A few applicable examples I could find (and all of which passed), are:

1. `Tests_Post_Query::test_the_posts_filter()`
2. `Tests_Post_GetPages::test_get_pages_cache()` et al. (why didn't this one catch the bug this ticket addresses?)
3. Everything in `Tests_Post_GetPosts`.

Is there a scenario you'd like to have explicitly tested, which isn't covered already in many other tests?

**Comment by peterwilsoncc** [comment]
@sybrew I hope you don't mind but I have retargeted this to trunk and pushed some tests to your repo in https://github.com/WordPress/wordpress-develop/pull/1806/commits/b05ce534851e9bf7271518db3fc457ca64a4f6e3

File: tests/phpunit/tests/post/updatePostCache.php [hunk 1]
```diff
@@ -0,0 +1,120 @@
...
+		self::$post_ids = $factory->post->create_many( 1 );
+	}
+
+	/**
+	 * Set up test resources before each test.
+	 */
+	public function set_up() {
+		wp_cache_set( 'last_changed', microtime(), 'posts' );
+	}
+
```
Comment (peterwilsoncc) [inline tests/phpunit/tests/post/updatePostCache.php]:
```suggestion
```

Pointless busy work, dealt with in the test suite 

https://github.com/WordPress/wordpress-develop/blob/8a4ffe388300701a0205d7d3ca9ea8fa4e861031/tests/phpunit/includes/abstract-testcase.php#L195-L200

File: tests/phpunit/tests/post/updatePostCache.php [hunk 2]
```diff
@@ -0,0 +1,120 @@
...
+ * Test `wp_update_cache()`.
+ *
+ * @package WordPress
+ */
+
+/**
+ * Test class for `wp_update_cache()`.
+ *
+ * @ticket 50567
+ *
```
Comment (peterwilsoncc) [inline tests/phpunit/tests/post/updatePostCache.php]:
```suggestion
```

Further tests may be added that don't relate to this ticket.

Each of the tests include this in their docblock so no practical affect for now.

**Review by costdev — APPROVED** [review]

File: tests/phpunit/tests/post/updatePostCache.php [hunk 3]
```diff
@@ -0,0 +1,111 @@
...
+	 * @ticket 50567
+	 */
+	public function test_get_post_caches_post_filter_with_object() {
+		$post_id = self::$post_ids[0];
+		$post    = (object) array( 'ID' => $post_id );
+
+		get_post( $post );
+
+		$cached_post = wp_cache_get( $post_id, 'posts' );
+		$this->assertSame( 'raw', $cached_post->filter );
```
Comment (costdev) [inline tests/phpunit/tests/post/updatePostCache.php]: To stabilise the test:
```suggestion
		$this->assertIsObject(
			$cached_post,
			'The cached post is not an object'
		);
		
		$this->assertObjectHasAttribute(
			'filter',
			$cached_post,
			'The cached post does not have a "filter" property'
		);
		
		$this->assertSame(
			'raw',
			$cached_post->filter,
			'The filter is not set to "raw"'
		);
```

**Comment by peterwilsoncc** [comment]
Committed in https://core.trac.wordpress.org/changeset/53042

**Comment by sybrew** [comment]
Committed via https://github.com/WordPress/wordpress-develop/commit/b1b305b82631ab2c7d72f0388cda035c8454f1e5. 

Somewhat sad, the lines I contributed directly don't have my blame attached; why was this decision made?

**Comment by peterwilsoncc** [comment]
WP is actually committed via SVN on `develop.svn.wordpress.org`, what's shown in GitHub is a mirror of that. 

Unfortunately that means that the `git blame` only shows the core committer who did the act of committing, rather than the additional information GitHub provides. There's a current discussion on how to improve that. https://make.wordpress.org/core/2022/01/31/ensuring-proper-attribution-for-contributions-to-wordpress-on-github/

In terms of including you in the credits for WP, they're generated from the `props` line in the commit message which includes your wordpress.org username.

**Comment by sybrew** [comment]
Thank you for the clarification. 

I'm glad this is finally merged into Core, and I'm eager to hear of improvements -- a patch that implemented this fix I shared among friends astounded them at how fast their sites have become on cheap hosting.

With all other resolved performance tickets combined, WP 6.0 seems to be on the right track to becoming "the update for the planet." Thank you for pushing that effort.

**Pull request closed without merging**